docs/en/lazy.md

```diff
@@ -1,16 +1,86 @@
 # `lazy()`
 
 Creates a [`loadable()`](./loadable.md) component which is loaded automatically when it is rendered for the first time.
+Basically, this utility calls `.load()` method or a loadable component automatically, when it is being rendered for the first time.
 
 Use it exactly the same as `loadable()`, the only difference is that it will automatically `.load()` on first render.
 
-However, you still have access to the `.load()` method, which allows you *"pre-load"* your components ahead of time.
+However, you still have access to the `.load()` method, which allows you *"pre-load"* your components ahead of time, if necessary.
+This ways &mdash; when your component is rendered for the first time &mdash; you will not experience a network delay.
 
 
-## Reference
+## Example
 
-```ts
-lazy: (params: ILazyParams) => React.ComponentClass;
+Below `MyComponent` will be resolved and renderer in place of `LazyMyComponent`.
 
-interface ILazyParams extends ILoadableParams {}
+```jsx
+const loader = () => new Promise(resolve => resolve(MyComponent));
+
+const LazyMyComponent = lazy({loader});
+
+<LazyMyComponent />
+```
+
+This feature allows you to do code-splitting &mdash; to load your components only when they are actually renderer for the first time.
+Below code will load and render `MyComponent` in place of `LazyMyComponent`, but only when `LazyMyComponent` was actually renderer for
+the first time.
+
+```jsx
+const loader = () => import('./MyComponent'); // Assume it is exports as default exports.
+
+const LazyMyComponent = lazy({loader});
+
+<LazyMyComponent />
+```
+
+Above code assumes your component was exported as default exports, if its not you would modify your code as so:
+
+```jsx
+const loader = () => import('./MyComponent').then(module => module.MyComponent);
+
+const LazyMyComponent = lazy({loader});
+
+<LazyMyComponent />
+```
+
+You can also use `.load()` method, to pre-load your component ahead of time.
+
+```js
+LazyMyComponent.load();
 ```
+
+By default, while your component is loading, nothing will be renderer, but you can display a loading text instead.
+
+```js
+const LazyMyComponent = lazy({
+  loader,
+  loading: 'MyComponent is loading...'
+});
+```
+
+Or any other React component:
+
+```js
+const LazyMyComponent = lazy({
+  loader,
+  loading: <LoadingSpinner />
+});
+```
+
+If error happens during loading, and `.error()` param is called, it receives a single error argument, and expects back
+a component to be renderer on error, or `null`.
```

Comment:
Typo: `rendered`.

Reply:
I guess I know, because for `nano-css`, I had to type `renderer` a lot.